src/main/java/com/gzs/main/App.java

```diff
@@ -19,22 +21,29 @@ public static void main(String[] args) throws Exception {
 //        generateDBData();
 
         log.info("Starting application");
+        config = new Configuration();
+
         Runtime.getRuntime().addShutdownHook(new Thread(() -> {
-            LanguageDaoImpl.endStatements();
-            TermDaoImpl.endStatements();
-            TranslationDaoImpl.endStatements();
+            log.info("Starting application shutdown process.");
```

Comment:
Idea is to be able to choose the DAO implementation (MySQL or InMemory) in runtime. This way you're hardcoding the dependency to MySQL one, so it wouldn't work with InMemory implementation.

One way to sort this is to add some cleanup method (`close` for example) in the interface and implement it appropriately in each Dao (MySQL will close statements, and InMemory would probably do nothing).

Reply:
@celicni Added closeResources method to DaoFactory.class. Now,when closing App,that method,based on data type defined in configuration,is called.